src/core/or/circuitpadding.h

```diff
@@ -92,6 +92,15 @@ typedef uint32_t circpad_delay_t;
  */
 #define CIRCPAD_INFINITY_BIN(mi)  ((mi)->histogram_len-1)
 
+/**
+ * Macro to check if we have mutable histograms in circpad_machine_runtime_t.
+ *
+ * Having hutable histograms in the circpad_machine_runtime_t is equivalent to
+ * circpad_state_t.token_removal != CIRCPAD_TOKEN_REMOVAL_NONE.
+ */
+#define CIRCPAD_IS_TOKEN_REMOVAL_SUPPORTED(mi) \
```

Comment:
Since this macro uses mi twice, it is not perfectly safe. I'd suggest a static inline function instead.

Reply:
Ok this is much better all around: 28a081dacca387d160ae02eda3a4176610e0ff12 (in PR #1024).

Thanks.